hebsafeharbor/common/date_regex.py

```diff
@@ -2,9 +2,23 @@
 heb_months_re = '|'.join(["אלול","אב","תמוז","סיון","סיוון","אייר","ניסן","אדר","שבט","טבת","כסלו","כיסלו","חשוון","תשרי"])
 heb_days_re = '|'.join(["[א-ל]׳","""י״(?:[א-ד]|[ז-ט])""","""ט״(?:ו|ז)""","""כ״[א-ט]""","""ל״א"""])
 heb_year_re = r"ה?ת[א-ת][א-ת]?״[א-ת]"
-HEB_DATE_REGEX =  rf"\b(?:ב|ל|מ|מה|ה)?({heb_days_re})?(?:\s+)?(?:ב|ל|בחודש|לחודש)?(?:-|\s)?({heb_months_re})(?:\s|,\s)?({heb_year_re})?\b"
+HEB_FULL_DATE_REGEX =  rf"\b(?:ב|ל|מ|מה|ה)?(?P<day>{heb_days_re})\s+(?:ב|ל|בחודש|לחודש)?(?:-|\s)?(?P<month>{heb_months_re})(?:\s+|,\s+)(?P<year>{heb_year_re})\b"
+HEB_MONTH_YEAR_REGEX =  rf"\b(?:ב|ל|בחודש|לחודש)?(?:-|\s)?(?P<month>{heb_months_re})(?:\s+|,\s+)(?P<year>{heb_year_re})\b"
+HEB_DAY_MONTH_REGEX =  rf"\b(?:ב|ל|מ|מה|ה)?(?P<day>{heb_days_re})\s+(?:ב|ל|בחודש|לחודש)?(?:-|\s)?(?P<month>{heb_months_re})\b"
 
 # Latin Dates
 latin_months_re =    '|'.join(["ינואר","פברואר","מרץ","אפריל","מאי","יוני","יולי","אוגוסט","ספטמבר","אוקטובר","נובמבר","דצמבר"])
 num_days_re = r"[1-9]|0[1-9]|[1-2][0-9]|3[0-1]"
-LATIN_DATE_REGEX =  rf"\b(?:ב|ל|מ|מה|ה)?({num_days_re})?(?:\s+)?(?:ב|ל|בחודש|לחודש)?(?:-|\s)?({latin_months_re})(?:\s|,\s)?(\d\d|\d\d\d\d)?\b"
\ No newline at end of file
+LATIN_DATE_REGEX =  rf"\b(?:ב|ל|מ|מה|ה)?(?P<day>{num_days_re})?(?:\s+)?(?:ב|ל|בחודש|לחודש)?(?:-|\s)?(?P<month>{latin_months_re})(?:\s|,\s)?(?P<year>\d\d|\d\d\d\d)?\b"
```

Comment:
please define the pattern (?:ב|ל|מ|מה|ה)? as a variable and use it in lines 5,6,7,12,21,22,23

Reply:
please define the pattern (?:ב|ל|בחודש|לחודש) as variable and use it in lines 5, 7, 12